### PR TITLE
(fix) O3-1594 BUG: Patient chart close button has inconsistent behavior

### PR DESCRIPTION
**Pull request by brandones**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Summary

Uses the new history utilities in esm-core to fix the patient-chart close button.

## Screenshots

The videos are 90% loading screens and are hard to follow because the cursor doesn't show up. So I've described them below.

Before: I access Lucy Karl's chart from patient-lists, refresh the page, and hit close. This takes me to the Home page. Then I go back to patient-lists, ctrl-click Lucy Karl's chart, change to that tab, and hit close. Nothing happens.

https://github.com/openmrs/openmrs-esm-patient-chart/assets/1031876/45ce0141-67ae-4c0c-a8c8-4bf1858bf0c6

After: I access Lucy Karl's chart from patient-lists, refresh the page, and hit close. This takes me back to patient-lists. Then I ctrl-click Lucy Karl's chart, change to that tab, and hit close. It takes me back to patient-lists in that new tab.

https://github.com/openmrs/openmrs-esm-patient-chart/assets/1031876/110d7c19-6f94-4220-8ae5-8cdf5d1c79d7

## Related Issue

https://openmrs.atlassian.net/browse/O3-1594

### CODE REVIEW

**Comment by brandones** [comment]
This should fix the test failures: https://github.com/openmrs/openmrs-esm-core/pull/886

**Review by vasharma05 — APPROVED** [review]
